lib/vuln/wp_xmlrpc_dos/engine.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author Aman Gupta; github.com/aman566
+
+import socket
+import threading
+import time
+import string
+import random
+import os
+import json
+import requests
+import socks
+from core.alert import messages, info, warn
+from core.targets import target_type
+from core.targets import target_to_host
+from core.load_modules import load_file_path
+from lib.socks_resolver.engine import getaddrinfo
+from core._time import now
+from core.log import __log_into_file
+from lib.payload.wordlists.useragents import useragents
+
+def extra_requirements_dict():
+    return {
+        "wp_xmlrpc_dos_vuln_ports": [80, 443]
+    }
+
+
+def check(target, port, headers, timeout_sec, log_in_file, language, retries, time_sleep, thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    time.sleep(time_sleep)
+    try:
+        if socks_proxy is not None:
+            socks_version = socks.SOCKS5 if socks_proxy.startswith(
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(
+                    socks_proxy.rsplit(':')[0]), int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+        _n = 0
+        while 1:
+            try:
+                if target_type(target) != "HTTP" and port == 443:
+                    target = 'https://' + target
+                if target_type(target) != "HTTP" and port == 80:
+                    target = 'http://' + target
+                target = target + '/xmlrpc.php'
+                postdata = '''<?xml version="1.0" encoding="utf-8"?><methodCall><methodName>system.listMethods</methodName><params></params></methodCall>'''
+                _r = requests.post(target, timeout=timeout_sec, headers=headers, data=postdata)
```

Comment:
@aman566 please add verify=False here as the request fails otherwise

Reply:
my bad. I will do it